server/src/main/webapp/WEB-INF/rails/webpack/models/pipeline_configs/pipeline_config.ts

```diff
@@ -281,4 +281,20 @@ export class Materials extends Array<Material> {
       this.splice(index, 1);
     }
   }
+
+  scmMaterialsHaveDestination(): boolean {
+    const hasScmMaterialWithEmptyDestination          = this
+      .filter((material) => material.attributes() instanceof ScmMaterialAttributes)
+      .some((material) => {
+        const attrs = material.attributes() as ScmMaterialAttributes;
+        return _.isEmpty(attrs.destination());
+      });
+    const hasPluggableScmMaterialWithEmptyDestination = this
+      .filter((material) => material.attributes() instanceof PluggableScmMaterialAttributes)
+      .some((material) => {
+        const attrs = material.attributes() as PluggableScmMaterialAttributes;
+        return _.isEmpty(attrs.destination());
+      });
```

Comment:
this would iterate the collection twice.
How about moving the `material.attributes() instanceof ScmMaterialAttributes` check inside `some`?